src/projects/digital/api/circuit/src/internal/assembly/components/ConstantNumberAssembler.ts

```diff
@@ -0,0 +1,118 @@
+import {V} from "Vector";
+
+import {Schema} from "shared/api/circuit/schema";
+
+import {Signal} from "digital/api/circuit/internal/sim/Signal";
+import {DigitalSim} from "digital/api/circuit/internal/sim/DigitalSim";
+import {AssemblerParams, AssemblyReason} from "shared/api/circuit/internal/assembly/Assembler";
+import {DigitalComponentInfo} from "../../DigitalComponents";
+import {ComponentAssembler} from "shared/api/circuit/internal/assembly/ComponentAssembler";
+import {FontStyle, Style} from "shared/api/circuit/internal/assembly/Style";
+
+
+export class ConstantNumberAssembler extends ComponentAssembler {
+    protected readonly sim: DigitalSim;
+
+    protected info: DigitalComponentInfo;
+
+    public constructor(params: AssemblerParams, sim: DigitalSim) {
+        super(params, V(1, 1), {
+            "outputs": (index) => this.getOutputLocations(index),
+        }, [
+            {
+                kind: "BaseShape",
+
+                dependencies: new Set([AssemblyReason.TransformChanged, AssemblyReason.SelectionChanged]),
+                assemble:     (comp) => this.assembleRectangle(comp),
+
+                getStyle: (comp) => this.options.fillStyle(this.isSelected(comp.id)),
+            },
+            {
+                kind: "BaseShape",
+
+                dependencies: new Set([AssemblyReason.TransformChanged, AssemblyReason.SelectionChanged]),
+                assemble:     (comp) => this.assembleLine(comp),
+
+                getStyle: (comp) => this.getLineStyle(comp),
+            },
+            {
+                kind: "Text",
+
+                dependencies: new Set([AssemblyReason.TransformChanged, AssemblyReason.SelectionChanged, AssemblyReason.PropChanged]),
+                assemble:     (comp) => this.assembleText(comp),
+
+                getFontStyle: () => this.getFontStyle(),
+            },
+        ]);
+        this.sim = sim;
+        this.info = this.circuit.getComponentInfo("Switch").unwrap() as DigitalComponentInfo;
+    }
+
+    private getOutputLocations(index: number) {
+        const targetY = .75 - (index * 1.5 / 3);
+        return { origin: V((this.size.x - this.options.defaultBorderWidth) / 2, targetY), target: V(1.2, targetY) } as const
+    }
+
+    private assembleRectangle(comp: Schema.Component) {
+        const transform = this.getTransform(comp).copy();
+        transform.setSize(transform.getSize().sub(V(this.options.defaultBorderWidth)));
```

Comment:
nit: mutability is gross, just do `const transform = new Transform(this.getPos(comp), this.getAngle(comp), this.getSize(comp).sub(V(this.options.defaultBorderWidth))` --- but actually, why do you need to subtract the border width? in the constant low/high you changed the port's origin instead? (add a comment if you think it's still necessary)

Reply:
This isn't really about getting ports to line up. Not including the border, the constant number should be the same size as constant high/low.
nightly:
![image](https://github.com/user-attachments/assets/12902724-238d-46c4-82d1-361cdb882dab)
without subtraction:
![image](https://github.com/user-attachments/assets/b28258d1-7cde-4d49-8d9f-7997768d4ada)
with subtraction:
![image](https://github.com/user-attachments/assets/32635247-025e-4aa6-adf8-9c32016d6641)